tests/components/control4/test_config_flow.py

```diff
@@ -0,0 +1,211 @@
+"""Test the Control4 config flow."""
+import asyncio
+import datetime
+import sys
+
+from pyControl4.error_handling import Unauthorized
+
+from homeassistant import config_entries, setup
+from homeassistant.components.control4.const import DEFAULT_SCAN_INTERVAL, DOMAIN
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_SCAN_INTERVAL,
+    CONF_USERNAME,
+)
+
+from tests.async_mock import AsyncMock, patch
+from tests.common import MockConfigEntry
+
+
+def _get_mock_c4_account(
+    getAccountControllers={
+        "controllerCommonName": "control4_model_00AA00AA00AA",
+        "href": "https://apis.control4.com/account/v3/rest/accounts/000000",
+        "name": "Name",
+    },
+    getDirectorBearerToken={
+        "token": "token",
+        "token_expiration": datetime.datetime(2020, 7, 15, 13, 50, 15, 26940),
+    },
+):
+    c4_account_mock = AsyncMock()
+
+    if sys.version_info[:2] < (3, 8):
```

Comment:
This shouldn't be needed. Try speccing the mock with the class. Use the `spec=` keyword. That should set the correct awaitable mock methods.

https://docs.python.org/3/library/unittest.mock.html#unittest.mock.AsyncMock